### PR TITLE
build: Update dependencies

### PR DESCRIPTION
**Pull request by AnnAngela**
As https://github.com/wdhongtw/vscode-gpg-indicator/pull/46#issuecomment-1449671426 said, this is the separate pr for the chore job. @wdhongtw

### CODE REVIEW

**Comment by AnnAngela** [comment]
@wdhongtw The development is nearing completion, I will test the new codes this night on my local machine.

If we can take a look at this pr and merge it, I think I can make a new pr tomorrow.

File: package.json [hunk 1]
```diff
@@ -1,81 +1,79 @@
...
-	"engines": {
-		"vscode": "^1.45.0"
-	},
-	"categories": [
-		"Other"
-	],
-	"icon": "images/icon-color.png",
-	"activationEvents": [
-		"workspaceContains:/.git",
-		"onCommand:gpgIndicator.unlockCurrentKey"
```
Comment (wdhongtw) [inline package.json:26]: One activation event is dropped, is this change intended?

File: package.json [hunk 2]
```diff
@@ -1,81 +1,79 @@
...
-		"eslint-plugin-tsdoc": "^0.2.14",
-		"typescript": "^3.8.3"
-	},
-	"license": "MIT",
-	"repository": {
-		"type": "git",
-		"url": "https://github.com/wdhongtw/vscode-gpg-indicator"
-	},
-	"preview": false,
-	"dependencies": {}
```
Comment (wdhongtw) [inline package.json:80]: Is the empty `dependencies` map must be deleted?

File: .eslintrc.json [hunk 3]
```diff
@@ -17,7 +17,8 @@
         "no-throw-literal": "warn",
         "semi": "error",
         "no-tabs": "error",
-        "indent": "error",
+        "indent": "off",
+        "@typescript-eslint/indent": "error",
         "tsdoc/syntax": "warn"
     }
-}
+}
```
Comment (wdhongtw) [inline .eslintrc.json:24]: Should include EOL here.

**Comment by wdhongtw** [comment]
@AnnAngela

`package.json` `package-lock.json` and `eslintrc.json` all have a bunch of whitespace changes (from tab to space).

These are introduced by auto-formatting feature I think? I would like to avoid unnecessary line diffs.

File: package.json (hunk 1 above)
Comment (AnnAngela) [reply package.json:26]: Yes, it's the suggestion from VSCode:

image

File: package.json (hunk 2 above)
Comment (AnnAngela) [reply package.json:80]: It's deleted by the upgradation from lockfile v1 to v3 by npm.

In the next pr, I will add a new dependency, so at that time the `dependencies` property will be add back.

File: package.json (hunk 1 above)
Comment (AnnAngela) [reply package.json:26]: I recheck the manual, [the auto generation for `activationEvents` is only available from vscode 1.74](https://code.visualstudio.com/api/references/activation-events#onCommand), I will add it back **in the next pr**, sorry for that.

**Comment by AnnAngela** [comment]
> @AnnAngela
> 
> `package.json` `package-lock.json` and `eslintrc.json` all have a bunch of whitespace changes (from tab to space).
> 
> These are introduced by auto-formatting feature I think? I would like to avoid unnecessary line diffs.

No, it's done by npm itself automatically. And I think this is a one-time change, sorry for the inconvenience.

File: package.json (hunk 2 above)
Comment (wdhongtw) [reply package.json:80]: OK :D

File: package.json (hunk 1 above)
Comment (wdhongtw) [reply package.json:26]: OK :D

**Comment by wdhongtw** [comment]
looks good to me :D
